dashboard/app/public_json_api.go

```diff
@@ -170,3 +177,141 @@ func GetJSONDescrFor(page interface{}) ([]byte, error) {
 	}
 	return json.MarshalIndent(res, "", "\t")
 }
+
+type coveredBlock struct {
+	FromLine int `json:"from_line"`
+	FromCol  int `json:"from_column"`
+	ToLine   int `json:"to_line"`
+	ToCol    int `json:"to_column"`
+}
+
+type functionCoverage struct {
+	FuncName     string          `json:"func_name"`
+	Instrumented int             `json:"total_blocks"`
+	Blocks       []*coveredBlock `json:"covered_blocks"`
+}
+
+type fileCoverage struct {
+	FilePath  string              `json:"file_path"`
+	Functions []*functionCoverage `json:"functions"`
+}
+
+func writeExtAPICoverageFor(ctx context.Context, w io.Writer, ns, repo string) error {
+	// By default, return the previous month coverage. It guarantees the good numbers.
+	//
+	// The alternative is to return the current month.
+	// The numbers will jump every day, on the 1st date may drop down.
+	tps, err := coveragedb.GenNPeriodsTill(1, civil.DateOf(time.Now()).AddDays(-31), "month")
+	if err != nil {
+		return fmt.Errorf("coveragedb.GenNPeriodsTill: %w", err)
+	}
+	defaultTimePeriod := tps[0]
+	covDBClient := GetCoverageDBClient(ctx)
+	ff, err := coveragedb.MakeFuncFinder(ctx, covDBClient, ns, defaultTimePeriod)
+	if err != nil {
+		return fmt.Errorf("coveragedb.MakeFuncFinder: %w", err)
+	}
+	covCh, errCh := coveragedb.FilesCoverageStream(ctx, covDBClient, ns, defaultTimePeriod)
+
+	gzw := gzip.NewWriter(w)
+	defer gzw.Close()
+
+	jsonHeader := "{\n\t\"files\": [\n"
+	if _, err := gzw.Write([]byte(jsonHeader)); err != nil {
+		return fmt.Errorf("w.Write(header): %w", err)
+	}
+
+	enc := json.NewEncoder(gzw)
+	enc.SetIndent("\t\t", "\t")
+
+	commit := ""
+	funcWriterErrCh := make(chan error, 1)
+	go func() {
```

Comment:
Do you really need this goroutine? You start it and wait for its result right after that, it's essentially equivalent to just doing it synchronously.

Reply:
It allows to use 'return' to stop for-switch.

Reply:
Maybe it could become a separate function then?

Alternatively, does it work with labels like this?
```
outer:
for {
  select {
    case a := <-b:
      break outer
  }
}

Reply:
I'll change go routine to the function.